packages/core/src/routes-me/user-assets.ts

```diff
@@ -0,0 +1,103 @@
+import { readFile } from 'node:fs/promises';
+
+import { consoleLog } from '@logto/cli/lib/utils.js';
+import {
+  userAssetsServiceStatusGuard,
+  allowUploadMimeTypes,
+  maxUploadFileSize,
+  type UserAssets,
+  userAssetsGuard,
+  adminTenantId,
+} from '@logto/schemas';
+import { generateStandardId } from '@logto/shared';
+import { format } from 'date-fns';
+import { object } from 'zod';
+
+import RequestError from '#src/errors/RequestError/index.js';
+import koaGuard from '#src/middleware/koa-guard.js';
+import type { RouterInitArgs } from '#src/routes/types.js';
+import SystemContext from '#src/tenants/SystemContext.js';
+import assertThat from '#src/utils/assert-that.js';
+import { uploadFileGuard } from '#src/utils/storage/consts.js';
+import { buildUploadFile } from '#src/utils/storage/index.js';
+
+import type { AuthedMeRouter } from './types.js';
+
+/**
+ * Duplicated from `/user-assets` management API and used specifically for admin tenant.
```

Comment:
Maybe this two routes can share some logic, an refactor would be great.

Reply:
Added a task and will refactor it later.
https://linear.app/silverhand/issue/LOG-8881/refactor-the-code-in-user-assets-api-and-meuser-assets